MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/javaservices/GeocodingActivity.java

```diff
@@ -0,0 +1,263 @@
+
+    // Mapbox access token is configured here. This needs to be called either in your application
+    // object or in the same activity which contains the mapview.
+    Mapbox.getInstance(this, getString(R.string.access_token));
+
+    // This contains the MapView in XML and needs to be called after the access token is configured.
+    setContentView(R.layout.activity_javaservices_geocoding);
+
+    mapView = findViewById(R.id.mapView);
+    mapView.onCreate(savedInstanceState);
+    mapView.getMapAsync(this);
+  }
+
+  @Override
+  public void onMapReady(MapboxMap mapboxMap) {
+    this.mapboxMap = mapboxMap;
+    initTextViews();
+    initButtons();
+  }
+
+  private void initTextViews() {
+    latEditText = findViewById(R.id.geocode_latitude_editText);
+    longEditText = findViewById(R.id.geocode_longitude_editText);
+    geocodeResultTextView = findViewById(R.id.geocode_result_message);
+  }
+
+  private void initButtons() {
+    mapCenterButton = findViewById(R.id.map_center_button);
+    startGeocodeButton = findViewById(R.id.start_geocode_button);
+    chooseCityButton = findViewById(R.id.choose_city_spinner_button);
+    startGeocodeButton.setOnClickListener(new View.OnClickListener() {
+      @Override
+      public void onClick(View view) {
+        // Make sure the EditTexts aren't empty
+        if (TextUtils.isEmpty(latEditText.getText().toString())) {
+          latEditText.setError(getString(R.string.fill_in_a_value));
+        } else if (TextUtils.isEmpty(longEditText.getText().toString())) {
+          longEditText.setError(getString(R.string.fill_in_a_value));
+        } else {
+          if (latCoordinateIsValid(Double.valueOf(latEditText.getText().toString()))
+            && longCoordinateIsValid(Double.valueOf(longEditText.getText().toString()))) {
+            // Make a geocoding search with the values inputted into the EditTexts
+            makeGeocodeSearch(new LatLng(Double.valueOf(latEditText.getText().toString()),
+              Double.valueOf(longEditText.getText().toString())));
+          } else {
+            Toast.makeText(GeocodingActivity.this, R.string.make_valid_lat, Toast.LENGTH_LONG).show();
+          }
+        }
+      }
+    });
+    chooseCityButton.setOnClickListener(new View.OnClickListener() {
+      @Override
+      public void onClick(View view) {
+        showCityListMenu();
+      }
+    });
+    mapCenterButton.setOnClickListener(new View.OnClickListener() {
+      @Override
+      public void onClick(View view) {
+        // Get the map target's coordinates
+        Double targetLat = mapboxMap.getCameraPosition().target.getLatitude();
+        Double targetLong = mapboxMap.getCameraPosition().target.getLongitude();
+
+        // Fill the coordinate EditTexts with the target's coordinates
+        setCoordinateEditTexts(mapboxMap.getCameraPosition().target);
+
+        // Make a geocoding search with the target's coordinates
+        makeGeocodeSearch(new LatLng(targetLat, targetLong));
```

Comment:
This is little thing. You keep on getting the target. 
It might be easier to get target: target = mapboxMap.getCameraPosition().target;
and then just call 
makeGeocodeSearch(new LatLng(target));

Reply:
Right. Fixed